proco/connection_statistics/api.py

```diff
@@ -54,16 +54,19 @@ def get(self, request, *args, **kwargs):
 
 class CountryWeekStatsAPIView(RetrieveAPIView):
     permission_classes = (AllowAny,)
-    serializer_class = CountryWeeklyStatusSerializer
+    serializer_class = CountryWeeklyStatusGraphSerializer
 
     def get_object(self, *args, **kwargs):
         week = self.kwargs['week']
         year = self.kwargs['year']
         date = datetime.strptime(f'{year}-W{week}-1', '%Y-W%W-%w')
         instance = CountryWeeklyStatus.objects.filter(country_id=self.kwargs['country_id'], date__lte=date).last()
+
         if not instance:
             raise Http404
 
+        instance.previous_week_exists = CountryWeeklyStatus.objects.filter(country_id=self.kwargs['country_id'],
```

Comment:
this code can be safely moved to the `else` part of get_previous_week_exists. let's try to keep code clean when possible.

def get_previous_week_exists(self, obj):
   if not obj:
       return False

   if hasattr(obj, 'previous_week_exists'):
        return obj.previous_week_exists
    
   return CountryWeeklyStatus.objects.filter(country_id=obj.country_id, date__lte=obj.date-timedelta(days=7)).exists()